contrib/clojure-package/test/org/apache/clojure_mxnet/resource_scope_test.clj

```diff
@@ -0,0 +1,65 @@
+;;
+;; Licensed to the Apache Software Foundation (ASF) under one or more
+;; contributor license agreements.  See the NOTICE file distributed with
+;; this work for additional information regarding copyright ownership.
+;; The ASF licenses this file to You under the Apache License, Version 2.0
+;; (the "License"); you may not use this file except in compliance with
+;; the License.  You may obtain a copy of the License at
+;;
+;;    http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+
+(ns org.apache.clojure-mxnet.resource-scope-test
+  (:require [org.apache.clojure-mxnet.ndarray :as ndarray]
+            [org.apache.clojure-mxnet.symbol :as sym]
+            [org.apache.clojure-mxnet.resource-scope :as resource-scope]
+            [clojure.test :refer :all]))
+
+
+(deftest test-resource-scope-with-ndarray
+  (let [native-resources (atom {})
+        x (ndarray/ones [2 2])
+        return-val (resource-scope/using
```

Comment:
i have a wild idea. let me know if i'm way off here but `using` gives the impression that its form should be `(using x (do y))`. instead i think `using` the way you've defined it, is very much like a `let` IMHO. can the `return-val` be:
```clojure
(resource-scope/let 
  [temp-x (ndarray/ones [3 1])
   temp-y (ndarray/ones [3 1])]
  (swap! native-resources assoc :temp-x temp-x)
  (swap! native-resources assoc :temp-y temp-y)
  (ndarray/+ temp-x 1))
```

Reply:
I like where you are going - I think adding an alias for `let` is a good idea. It's also like a `do` though because any native resources created in the forms will be closed. What do you think about adding aliases for `resource-scope/let` and `resource-scope/do` while keeping `resource-scope/using` to keep with the Scala api?

Reply:
that sounds great!

Reply:
well `do` and `let` are special forms - so we need another name... I'm thing `resource-scope/with-do` and `resource-scope/with-let`

Reply:
I def. think `with-` is the right choice to convey resource cleanup.